src/internal/operations/manifests_test.go

```diff
@@ -959,8 +1039,19 @@ func (suite *OperationsManifestsUnitSuite) TestProduceManifestsAndMetadata_fallb
 			assert.False(t, b, "no-metadata is forced for this test")
 
 			manIDs := []string{}
+
 			for _, m := range mans {
 				manIDs = append(manIDs, string(m.ID))
+
+				reasons, ok := test.expectReasons[string(m.ID)]
+				assert.True(t, ok, "unexpected manifest in result: ", m.ID)
```

Comment:
nit: to reduce repeated messages in the test output it should be alright to skip the check on ok. If the map doesn't have the entry the assert on 1054 will report a mismatch. The assert on 1057 should report unexpected manifests
```suggestion
				reasons, ok := test.expectReasons[string(m.ID)]
```